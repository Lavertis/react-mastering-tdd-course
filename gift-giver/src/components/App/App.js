import {Component} from "react";
import {Button, Col} from "react-bootstrap";
import Gift from "../Gift/Gift";
import {maxNumber} from "../../helpers";

class App extends Component {
    constructor() {
        super();

        this.state = {gifts: []};
    }

    addGift = () => {
        const {gifts} = this.state;
        const ids = gifts.map(gift => gift.id);
        const maxId = maxNumber(ids);
        gifts.push({id: maxId + 1});
        this.setState({gifts});
    };

    removeGift = (id) => {
        const gifts = this.state.gifts.filter(gift => gift.id !== id);
        this.setState({gifts});
    };

    render() {
        return (
            <Col xs={4} className="App text-center mx-auto my-5">
                <h2>Gift Giver</h2>
                <div className="gift-list">
                    {this.state.gifts.map(gift => {
                        return (
                            <Gift
                                key={gift.id}
                                gift={gift}
                                removeGift={this.removeGift}
                            />
                        );
                    })}
                </div>
                <Button className="btn-add m-2" onClick={this.addGift}>Add Gift</Button>
            </Col>
        );
    }
}

export default App;
