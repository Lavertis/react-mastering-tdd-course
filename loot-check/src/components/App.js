function App() {
    return (
        <div className="App">
            <h2>Loot Check</h2>
        </div>
    );
}

export default App;
