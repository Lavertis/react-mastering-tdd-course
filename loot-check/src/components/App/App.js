import Wallet from "../Wallet/Wallet";

function App() {
    return (
        <div className="App">
            <h2>Loot Check</h2>
            <hr/>
            <Wallet/>
        </div>
    );
}

export default App;
