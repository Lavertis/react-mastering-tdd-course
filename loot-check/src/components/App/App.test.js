import {shallow} from "enzyme";
import App from "./App";

describe('App', function () {
    const app = shallow(<App/>);

    it('renders properly', function () {
        expect(app).toMatchSnapshot();
    });

    it('contains a connected Wallet component', function () {
        expect(app.find('Connect(Wallet)').exists()).toBeTruthy();
    });
});