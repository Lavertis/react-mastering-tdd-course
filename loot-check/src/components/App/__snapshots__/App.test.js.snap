// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
<div
  className="App"
>
  <h2>
    Loot Check
  </h2>
  <hr />
  <Connect(Wallet) />
</div>
`;
